pkg/compactor/blocks_cleaner.go

```diff
@@ -243,13 +243,23 @@ func (c *BlocksCleaner) loop(ctx context.Context) error {
 				continue
 			}
 			cleanJobTimestamp := time.Now().Unix()
-			usersChan <- &cleanerJob{
+
+			select {
+			case usersChan <- &cleanerJob{
 				users:     activeUsers,
 				timestamp: cleanJobTimestamp,
+			}:
+			default:
+				level.Warn(c.logger).Log("msg", "unable to push cleaning job to usersChan")
```

Comment:
I wonder if it is worth adding a counter metric of unable to push active user clean job to queue. Ideally it should work next retry but if current job is taking too long it is possible to drop those jobs for a long time.

The symptom of this would be bucket index update delay so it might not be hard to figure out the root cause from that without the metric. The metric is still nice to have though. I think we have similar metric in Query Frontend..

Reply:
I added metric. Also, improved the logs a bit